### PR TITLE
Initial support for identity if absent in JSON

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/IdentityIfAbsent.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.squareup.wire
```
Comment (oldergod) [inline wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/IdentityIfAbsent.kt:1]: copyright

Comment (oldergod) [reply wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/IdentityIfAbsent.kt:1]: hmm I noticed our `internal annotation class Uint64` doesn't have it either. Not needed for them?

Comment (swankjesse) [reply wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/IdentityIfAbsent.kt:1]: I like it for everything.

Comment (swankjesse) [reply wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/IdentityIfAbsent.kt:16]: Fixed x 2.

**Pull request merged**